pilot/pkg/xds/sds.go

```diff
@@ -270,9 +270,12 @@ func filterAuthorizedResources(resources []SecretResource, proxy *model.Proxy, s
 				deniedResources = append(deniedResources, r.Name)
 			}
 		case credentials.KubernetesSecretType:
+			// CA Certs are public information, so we allows allow these to be accessed (from the same namespace)
+			isCAOnlySecret := strings.HasSuffix(r.Name, securitymodel.SdsCaSuffix)
```

Comment:
This could be unsafe if a secret named with `-cacert ` but not root cert

Reply:
I thought the same but later on we explicitly only read root cert if the name is -cacert. So it should not be possible

Reply:
Let me add a test case to handle this though as it is a risk

Reply:
Ok, added a test to verify this as well.